app/gui/src/ide/initializer.rs

```diff
@@ -109,19 +106,17 @@ impl Initializer {
         match &self.config.backend {
             ProjectManager { endpoint } => {
                 let project_manager = self.setup_project_manager(endpoint).await?;
-                let project_name = self.config.project_name.clone();
-                let controller = controller::ide::Desktop::new_with_opened_project(
-                    project_manager,
-                    project_name,
-                )
-                .await?;
+                let maybe_project_name = self.config.project_name.clone();
+                let controller =
+                    controller::ide::Desktop::new(project_manager, maybe_project_name).await?;
                 Ok(Rc::new(controller))
             }
             LanguageServer { json_endpoint, binary_endpoint, namespace } => {
                 let json_endpoint = json_endpoint.clone();
                 let binary_endpoint = binary_endpoint.clone();
                 let namespace = namespace.clone();
-                let project_name = self.config.project_name.clone();
+                // unwrap is safe here, as BackendService constructor made this check
+                let project_name = self.config.project_name.clone().unwrap();
```

Comment:
Please refactor the code in such a way that it does not use unwrap. Making code "safe by comment" is very bad. Someone will refactor that, do not notice the comment and it will break. Do not use `unwrap` or other possible crashing functions until absolutely necessary.

Reply:
If we require a project name when giving ls connection (and this makes sense), then the name should be put inside `BackendService::LanguageServer`

Reply:
Done as proposed, now project name is a part of `LanguageServer`, unwrap removed.